repose-aggregator/services/atom-feed/impl/src/main/scala/org/openrepose/nodeservice/atomfeed/impl/AtomFeedServiceImpl.scala

```diff
@@ -0,0 +1,230 @@
+
+  @PostConstruct
+  def init(): Unit = {
+    logger.info("Initializing and registering configuration listeners")
+    val xsdURL = getClass.getResource("/META-INF/schema/config/atom-feed-service.xsd")
+
+    configurationService.subscribeTo(
+      DEFAULT_CONFIG,
+      xsdURL,
+      AtomFeedServiceConfigurationListener,
+      classOf[AtomFeedServiceConfigType]
+    )
+    configurationService.subscribeTo(
+      SYSTEM_MODEL_CONFIG,
+      SystemModelConfigurationListener,
+      classOf[SystemModel]
+    )
+  }
+
+  @PreDestroy
+  def destroy(): Unit = {
+    logger.info("Unregistering configuration listeners and shutting down service")
+    configurationService.unsubscribeFrom(DEFAULT_CONFIG, AtomFeedServiceConfigurationListener)
+    configurationService.unsubscribeFrom(SYSTEM_MODEL_CONFIG, SystemModelConfigurationListener)
+
+    actorSystem.shutdown()
+  }
+
+  override def registerListener(feedId: String, listener: AtomFeedListener): String = {
+    val listenerId = java.util.UUID.randomUUID().toString
+    logger.info("Attempting to register a listener with id: " + listenerId + " to feed with id: " + feedId)
+
+    val notifierManager = feedActors.get(feedId) match {
+      case Some(FeedActorPair(manager, _)) =>
+        manager
+      case None =>
+        val manager = actorSystem.actorOf(Props[NotifierManager], name = feedId + NOTIFIER_MANAGER_TAG)
+        feedActors = feedActors + (feedId -> FeedActorPair(manager, None))
+
+        if (isServiceEnabled) {
+          manager ! ServiceEnabled
+        }
+
+        manager
+    }
+
+    notifierManager ! AddNotifier(listenerId, listener)
+
+    listenerNotifierManagers = listenerNotifierManagers + (listenerId -> notifierManager)
+
+    listenerId
+  }
+
+  override def unregisterListener(listenerId: String): Unit = {
+    logger.info("Attempting to unregister the listener with id: " + listenerId)
+
+    listenerNotifierManagers.get(listenerId) match {
+      case Some(notifierManager) =>
+        notifierManager ! RemoveNotifier(listenerId)
+        listenerNotifierManagers = listenerNotifierManagers - listenerId
+      case None =>
+        logger.debug("Listener not registered: " + listenerId)
+    }
+  }
+
+  case class FeedActorPair(notifierManager: ActorRef, feedReader: Option[ActorRef])
+
+  private object AtomFeedServiceConfigurationListener extends UpdateListener[AtomFeedServiceConfigType] {
+    private var initialized = false
+
+    override def configurationUpdated(configurationObject: AtomFeedServiceConfigType): Unit = {
+      synchronized {
+        logger.trace("Service configuration updated")
+        initialized = true
+
+        configurationObject.getFeed foreach { feedConfig =>
+          val notifierManager = feedActors.get(feedConfig.getId) match {
+            case Some(actorPair) =>
+              // todo: only replace the current feed reader if the configuration has changed
```

Comment:
Kind of? I think?

Right now we don't keep track of the old config, so I don't have both when the config changes, but that's easy enough to change.

If we're only looking at a couple of fields, I think we can maintain the high water mark on a timing change.

Let me try to implement this and you can tell me what you think. It might be much easier than it sounds.